### PR TITLE
grpc proxy: revert backup target grpc call

### PR DESCRIPTION
**Pull request by c3y1huang**
https://github.com/longhorn/longhorn/issues/4076

### CODE REVIEW

File: controller/backup_controller.go [hunk 1]
```diff
@@ -266,15 +266,15 @@ func (bc *BackupController) reconcile(backupName string) (err error) {
 
 		if backupTarget.Spec.BackupTargetURL != "" &&
 			backupVolume != nil && backupVolume.DeletionTimestamp == nil {
-			engineClientProxy, backupTargetConfig, err := getBackupTarget(bc.controllerID, backupTarget, bc.ds, log)
+			engineClientProxy, backupTargetClient, err := getBackupTarget(bc.controllerID, backupTarget, bc.ds, log)
```
Comment (innobead) [inline controller/backup_controller.go:269]: Do we still need `getBackupTarget`? or just use `getBackupTargetClient` instead. I feel we don't need the returned `engineClientProxy` any more here.

**Review by innobead — CHANGES_REQUESTED** [review]

File: controller/backup_controller.go (hunk 1 above)
Comment (c3y1huang) [reply controller/backup_controller.go]: Fixed.

**Comment by c3y1huang** [comment]
Test result:
- test_backup: https://ci.longhorn.io/job/private/job/longhorn-tests-regression/1470/
- coretest: https://ci.longhorn.io/job/private/job/longhorn-tests-regression/1471/
- full regression: https://ci.longhorn.io/job/private/job/longhorn-tests-regression/1472/

**Review by shuo-wu — APPROVED** [review]

**Review by derekbit — APPROVED** [review]
LGTM

**Pull request merged**